src/testdrive/src/action/kafka/verify.rs

```diff
@@ -273,3 +274,98 @@ impl Action for VerifyAction {
+        [value] => (None, value.clone()),
+        [key, value] => (Some(key.clone()), value.clone()),
+        _ => return Err(String::from("unable to read key/value schema inputs"))
+    };
+
+    cmd.args.done()?;
+    Ok(VerifySchemaAction {
+        sink,
+        format,
+        expected_key_schema: key,
+        expected_value_schema: value,
+        context,
+    })
+}
+
+#[async_trait]
+impl Action for VerifySchemaAction {
+    async fn undo(&self, _state: &mut State) -> Result<(), String> {
+        Ok(())
+    }
+
+    async fn redo(&self, state: &mut State) -> Result<(), String> {
+        let topic = get_topic(&self.sink, "topic", state).await?;
+
+        match &self.format {
+            SinkFormat::Avro => {
+                let generated_value_schema = state
+                    .ccsr_client
+                    .get_schema_by_subject(&format!("{}-value", topic))
+                    .await
+                    .map_err(|e| format!("fetching schema: {}", e))?
+                    .raw;
+
+                let generated_key_schema = state
+                    .ccsr_client
+                    .get_schema_by_subject(&format!("{}-key", topic))
+                    .await
+                    .ok()
+                    .map(|key_schema| {
+                        avro::parse_schema(&key_schema.raw)
+                            .map_err(|e| format!("parsing avro schema: {}", e))
+                    })
+                    .transpose()?;
+
+                let generated_value_schema = avro::parse_schema(&generated_value_schema)
+                    .map_err(|e| format!("parsing generated avro schema: {}", e))?;
+                let expected_value_schema = avro::parse_schema(&self.expected_value_schema)
+                    .map_err(|e| format!("parsing expected avro schema: {}", e))?;
+
+                if expected_value_schema.ne(&generated_value_schema) {
+                    return Err(format!("value schema did not match\nexpected:\n{:?}\n\nactual:\n{:?}",
+                        expected_value_schema, generated_value_schema));
+                }
+
+                if let Some(expected_key_schema) = &self.expected_key_schema {
+                    let expected_key_schema = avro::parse_schema(expected_key_schema)
+                        .map_err(|e| format!("parsing expected avro schema: {}", e))?;
+
+                    if generated_key_schema.is_none() {
+                        return Err(String::from("empty generated key schema"));
+                    }
+
+                    let generated_key_schema = generated_key_schema.unwrap();
+                    if expected_key_schema.ne(&generated_key_schema) {
+                        return Err(format!("key schema did not match\nexpected:\n{:?}\n\nactual:\n{:?}",
+                                           expected_key_schema, generated_key_schema));
+                    }
+                }
+            },
+            _ => unimplemented!()
```

Comment:
Could you add an error message here, like `unimplemented!("kafka-verify-schema is only supported for Avro sinks")`